src/main/java/com/example/booksstore/service/impl/OrderServiceImpl.java

```diff
@@ -0,0 +1,83 @@
+package com.example.booksstore.service.impl;
+
+import com.example.booksstore.dto.order.OrderRequestDto;
+import com.example.booksstore.dto.order.OrderResponseDto;
+import com.example.booksstore.dto.order.UpdateOrderRequestDto;
+import com.example.booksstore.exceptions.EntityNotFoundException;
+import com.example.booksstore.mappers.OrderMapper;
+import com.example.booksstore.models.CartItem;
+import com.example.booksstore.models.Order;
+import com.example.booksstore.models.OrderItem;
+import com.example.booksstore.models.ShoppingCart;
+import com.example.booksstore.models.User;
+import com.example.booksstore.repository.OrderRepository;
+import com.example.booksstore.repository.ShoppingCartRepository;
+import com.example.booksstore.service.OrderItemService;
+import com.example.booksstore.service.OrderService;
+import jakarta.transaction.Transactional;
+import java.math.BigDecimal;
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.Set;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class OrderServiceImpl implements OrderService {
+    private final OrderRepository orderRepository;
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final OrderItemService orderItemService;
+    private final OrderMapper orderMapper;
+
+    @Transactional
+    @Override
+    public OrderResponseDto save(OrderRequestDto requestDto, User user) {
+        ShoppingCart shoppingCart = shoppingCartRepository
+                .findByUserId(user.getId())
+                .orElseThrow(() -> new EntityNotFoundException(
+                        "Can't find shopping cart for user " + user.getId()));
+
+        Set<CartItem> cartItems = shoppingCart.getCartItems();
+        if (cartItems.isEmpty()) {
+            throw new EntityNotFoundException("Can't find cart items for user " + user.getId());
+        }
+
+        shoppingCartRepository.deleteById(shoppingCart.getId());
+        Order order = orderMapper.toModel(requestDto);
+        order.setUser(user);
+        order.setStatus(Order.Status.COMPLETED);
+        order.setOrderDate(LocalDateTime.now());
+        order.setTotal(calculateTotal(order));
+        Order savedOrder = orderRepository.save(order);
+
+        orderItemService.saveOrderItems(savedOrder.getOrderItems());
```

Comment:
if we use cascades here - we'd have only 1 save request here

Reply:
Are you talking about CascadeType.PERSIST?

Reply:
yes, but I think we can even use `CascadeType.ALL` for orderItems in order entity, since it is basically a child entity of order

Reply:
now you can remove the second call and order items will be saved by the cascade of order save
```suggestion
        Order savedOrder = orderRepository.save(order);
```